### PR TITLE
Design for maps

### PR DESCRIPTION
**Pull request by vladminsky**

### CODE REVIEW

File: src/elements/coords.geomap.js [hunk 1]
```diff
@@ -157,81 +234,174 @@ export class GeoMap {
...
-                    value = guide.defaultFill;
-                } else if (codes.length > 0) {
-                    value = fillScale(groupByCode[props[codes[0]]]);
-                }
-
-                return value;
-            })
+        var xmap = node
+            .selectAll('.map-container')
+            .data([`${innerW}${innerH}${contours.join('-')}`], (x) => x);
```
Comment (Mavrin) [inline src/elements/coords.geomap.js]: _.indentity

File: src/elements/coords.geomap.js [hunk 2]
```diff
@@ -157,81 +234,174 @@ export class GeoMap {
...
+                var labelsHash = {};
+                var reverseContours = contours.reduceRight((m, t) => (m.concat(t)), []);
 
-            path.pointRadius(1.5);
+                if (guide.showNames) {
+                    labelsHash = self._calcLabels(topoJSONData, reverseContours, path);
+                }
 
-            var placesFeature = topojson.feature(topoJSONData, topoJSONData.objects.places);
+                var grayScale = ['#fbfbfb', '#fffefe', '#fdfdff', '#fdfdfd', '#ffffff'];
```
Comment (vladminsky) [inline src/elements/coords.geomap.js]: try use CSS for that

File: src/utils/utils-draw.js [hunk 3]
```diff
@@ -2,7 +2,25 @@
...
+        s2_x = bx1 - bx0;
+        s2_y = by1 - by0;
+
+        var s, t;
+        s = (-s1_y * (ax0 - bx0) + s1_x * (ay0 - by0)) / (-s2_x * s1_y + s1_x * s2_y);
+        t = ( s2_x * (ay0 - by0) - s2_y * (ax0 - bx0)) / (-s2_x * s1_y + s1_x * s2_y);
+
+        if (s >= 0 && s <= 1 && t >= 0 && t <= 1) {
+            // Collision detected
+            return 1;
```
Comment (Mavrin) [inline src/utils/utils-draw.js]: Why is not boolean?

File: src/elements/coords.geomap.js [hunk 4]
```diff
@@ -87,8 +92,91 @@ export class GeoMap {
...
     }
 
+    _calcLabels(topoJSONData, reverseContours, path) {
+
+        var innerW = this.W;
+        var innerH = this.H;
+
+        var labelsHashRef = {};
+
+        reverseContours.forEach((c) => {
```
Comment (Mavrin) [inline src/elements/coords.geomap.js]: Maybe better use reduce?

**Pull request merged**